core/src/main/java/org/elasticsearch/cluster/routing/IndexRoutingTable.java

```diff
@@ -128,6 +128,11 @@ boolean validate(MetaData metaData) {
                     throw new IllegalStateException("shard routing has an index [" + shardRouting.index() + "] that is different " +
                                                     "from the routing table");
                 }
+                if (shardRouting.active() &&
+                    indexMetaData.activeAllocationIds(shardRouting.id()).contains(shardRouting.allocationId().getId()) == false) {
+                        throw new IllegalStateException("active shard routing " + shardRouting + " has no corresponding entry in the " +
```

Comment:
Fair point.  I think we could put comments on this method as well as the `public boolean validate(MetaData metaData)` in this class that they are meant only for testing purposes.